### PR TITLE
containers: replace sed calls with dedicated function

### PR DESCRIPTION
**Pull request by asmorodskyi**
VR : http://autobot.qa.suse.de/tests/2347

### CODE REVIEW

**Review by pdostal — APPROVED** [review]

File: lib/containers/common.pm [hunk 1]
```diff
@@ -49,8 +49,8 @@ sub install_podman_when_needed {
             push(@pkgs, 'apparmor-parser')   if is_leap("=15.1");    # bsc#1123387
             zypper_call "in @pkgs";
         }
-        assert_script_run('podman info');
     }
+    assert_script_run('podman info');
```
Comment (b10n1k) [inline lib/containers/common.pm:53]: this is not relevant to the commit and also i wonder if there are corner cases where if is skipped but podman not installed. i would suggest to just leave it out of this PR

**Review by b10n1k — CHANGES_REQUESTED** [review]
just to be save and keep commit relevant, please remove change in nstall_podman_when_needed

File: lib/containers/common.pm (hunk 1 above)
Comment (asmorodskyi) [reply lib/containers/common.pm:53]: what "corner cases" ? if `which podman` return 0 whatever will fail here should be a bug which we need to catch

**Comment by asmorodskyi** [comment]
> just to be save and keep commit relevant, please remove change in nstall_podman_when_needed

ok I will split it into separate commit but I would like to keep it

File: lib/containers/common.pm (hunk 1 above)
Comment (pdostal) [reply lib/containers/common.pm:53]: This change might be mentioned in the commit or in the description of this pull request but I'm also OK with keeping it here.

Comment (b10n1k) [reply lib/containers/common.pm:53]: i cant think any. but i think you are right. Can you at least make it a separate commit?

**Comment by asmorodskyi** [comment]
ok I mess up something when amending commit :facepalm:

**Comment by asmorodskyi** [comment]
closing this one . please follow same changes here https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/12415

**Pull request closed without merging**